grovel/asdf.lisp

```diff
@@ -86,22 +86,25 @@
   (let* ((input-file (first (input-files op c)))
          (output-file (make-pathname :type (generated-lisp-file-type c)
                                      :defaults input-file))
-         (c-file (make-c-file-name output-file "__grovel")))
+         (c-file (make-c-file-name output-file "__grovel"))
+         (exe-file (make-exe-file-name c-file)))
     (list output-file
           c-file
-          (make-exe-file-name c-file))))
+          exe-file)))
+
 
 (defmethod perform ((op process-op) (c grovel-file))
-  (let* ((output-file (first (output-files op c)))
-         (input-file (first (input-files op c)))
-         (tmp-file (process-grovel-file input-file output-file)))
-    (rename-file-overwriting-target tmp-file output-file)))
+  (destructuring-bind (output-file c-file exe-file) (output-files op c)
+    (let* ((input-file (first (input-files op c)))
+           (absolute-cache-dir (absolute-cache-dir input-file (cache-dir-of c))))
+      (process-grovel-file* input-file output-file c-file exe-file absolute-cache-dir))))
```

Comment:
output-files has the names we computed above, so we just destructure them. 
`#'absolute-cache-dir` returns nil if the `(cache-dir-of c)` is nil

We call `process-grovel-file*` as `process-grovel-file` is the function version with the old signature